doc/source/whatsnew/v1.4.0.rst

```diff
@@ -50,10 +50,19 @@ Notable bug fixes
 
 These are bug fixes that might have notable behavior changes.
 
-.. _whatsnew_140.notable_bug_fixes.notable_bug_fix1:
+.. _whatsnew_140.notable_bug_fixes.inconsistent_date_string_parsing:
 
-notable_bug_fix1
-^^^^^^^^^^^^^^^^
+Inconsistent date string parsing
```

Comment:
can you also add to the documentation itself, e.g. somewhere in `timeseries.rst` is appropriate